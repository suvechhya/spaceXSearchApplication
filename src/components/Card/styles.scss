@import '../../styles/main.scss';
.card {
    margin:5px;
    background-color: $white;
    border-radius: 5px;
    padding: 20px;
    h3 {
        color: $primary-green;
    }
    &__image {
        background-color: $background-grey;
        display: flex;
        justify-content: center;
        img{
            height: 40%;
            width: 80%;
        }
    }

    &__section{
        margin: 10px 0;
    }

    @media screen and (min-width: $tablet-min) and (max-width: $tablet-max) {
        min-height: 450px;
    }

    @media screen and (min-width: $desktop-min) {
        min-height: 485px;
    }
}