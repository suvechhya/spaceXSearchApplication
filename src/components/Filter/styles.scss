@import '../../styles/main.scss';

.filter-btn{
    height: 35px;
    width: 100px;
    border: none;
    background-color: $secondary-green;
    font-size: 1.2em;
    border-radius: 5px;
    margin: 10px 15px;
    cursor: pointer;
    &.selected{
        background-color: $primary-green;
        color: $white;
    }
    @media screen and (min-width: $tablet-min) and (max-width: $tablet-max) {
        height: 20px;
        width: 40px;
        font-size: 0.5em;
        margin: 5px 5px;
    }
}