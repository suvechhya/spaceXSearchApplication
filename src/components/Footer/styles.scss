@import '../../styles/main.scss';
.footer{
    position: fixed;
    left: 0;
    bottom: 0;
    height: 2em;
    width: 100%;
    background-color: $background-grey;
    color: $black;
    display: flex;
    justify-content: center;
    align-items: center;
}