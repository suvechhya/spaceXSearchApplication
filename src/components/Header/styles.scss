@import '../../styles/main.scss';
.header{
    position: fixed;
    left: 0;
    top: 0;
    padding-left: 1em;
    width: 100%;
    background-color: $background-grey;
    color: $black;
    display: flex;
    justify-content: flex-start;
}