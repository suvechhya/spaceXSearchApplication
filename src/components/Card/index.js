import React from 'react';
import './styles.scss';

const Card = ({ item: {mission_id,
    mission_name,
    launch_landing,
    launch_success,
    launch_year,
    mission_patch }}) => {
  return (
    <div className="card">
        <div className="card__image">
            <img src={mission_patch} />
        </div>
        <h3 className="card__name">{mission_name}</h3>
        <div className="card__section card__mission-id">
            <strong>Mission Ids:</strong>
            <ul>
                {mission_id && mission_id.length > 0 && mission_id.map((id) => (
                    <li>{id}</li>
                ))}
            </ul>
        </div>
        <div className="card__section card__launch-year">
            <strong>Launch Year: </strong>
            <span>{launch_year}</span>
        </div>
        <div className="card__section card__successful-launch">
            <strong>Successful Launch: </strong>
            <span>{launch_success}</span>
        </div>
        <div className="card__section card__launch-landing">
            <strong>Successful Landing: </strong>
            <span>{launch_landing}</span>
        </div>
    </div>
  );
}

export default Card;
