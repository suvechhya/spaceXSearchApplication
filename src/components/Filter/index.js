import React from 'react';
import { string, func, bool } from 'prop-types';
import clsx from 'clsx';
import './styles.scss';

const Filter = ({ text, value, clickHandler, isSelected }) => {
  return (
    <button className={clsx("filter-btn", {'selected': isSelected})} onClick={() => clickHandler(value)}>{text}</button>
  );
}

Filter.props = {
  text: string.isRequired,
  value: string.isRequired,
  clickHandler: func.isRequired,
  isSelected: bool.isRequired,
}

export default Filter;
