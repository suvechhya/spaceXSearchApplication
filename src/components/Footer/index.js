import React from 'react';
import './styles.scss';

const Footer = () => {
  return (
    <footer className="footer">
        <span className="footer__text">
            <strong>Developed By : </strong>
            Suvechhya Banerjee
        </span>
    </footer>
  );
}

export default Footer;
