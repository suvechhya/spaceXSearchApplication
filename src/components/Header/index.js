import React from 'react';
import './styles.scss';

const Header = () => {
  return (
    <header className="header">
        <h1>SpaceX Launch Programs</h1>
    </header>
  );
}

export default Header;
