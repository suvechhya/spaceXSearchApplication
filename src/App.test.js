import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from './components/Header';
import Footer from './components/Footer';
import Loader from './components/Loader';
import Card from './components/Card';
import Filter from './components/Filter';

describe('Header', () => {
    it('should render correctly with no props', () => {
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });

    it('should have the text SpaceX Launch Programs', () => {
        const component = shallow(<Header />);
        expect(component.find("h1").contains("SpaceX Launch Programs")).toEqual(true);
    });
});

describe('Footer', () => {
    it('should render correctly with no props', () => {
        const component = shallow(<Footer />);
        expect(component).toMatchSnapshot();
    });
});

describe('Loader', () => {
    it('should render correctly with no props', () => {
        const component = shallow(<Loader />);
        expect(component).toMatchSnapshot();
    });
});

describe('Card', () => {
    it('should render correctly with given props', () => {
        const item = {
            mission_id: 1,
            mission_name: 'Test Mission',
            launch_landing: true,
            launch_success: true,
            launch_year: 2008,
            mission_patch: 'http://abc.com'
        };
        const component = shallow(<Card item={item} />);
        expect(component).toMatchSnapshot();
    });
});

describe('Filter', () => {
    
    it('should render correctly with given props', () => {
        const component = shallow(<Filter text="2007" value="2007" isSelected={true} clickHandler={clickFn} />);
        expect(component).toMatchSnapshot();
    });

    it('should be possible to activate button with Spacebar', () => {
        const blankFn = () => {};
        const component = mount(<Filter text="2007" value="2007" isSelected={true} clickHandler={blankFn} />);
        component
            .find('button')
            .simulate('keydown', { keyCode: 32 });
        expect(component).toMatchSnapshot();
        component.unmount();
    });

    const clickFn = jest.fn();
    it('props function should be callable', () => {
        const component = shallow(<Filter text="2007" value="2007" isSelected={true} clickHandler={clickFn} />);
        component
            .find('button')
            .simulate('click');
        expect(clickFn).toHaveBeenCalled();
    });
});

