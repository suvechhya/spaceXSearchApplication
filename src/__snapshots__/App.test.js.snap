// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly with given props 1`] = `
<div
  className="card"
>
  <div
    className="card__image"
  >
    <img
      src="http://abc.com"
    />
  </div>
  <h3
    className="card__name"
  >
    Test Mission
  </h3>
  <div
    className="card__section card__mission-id"
  >
    <strong>
      Mission Ids:
    </strong>
    <ul />
  </div>
  <div
    className="card__section card__launch-year"
  >
    <strong>
      Launch Year: 
    </strong>
    <span>
      2008
    </span>
  </div>
  <div
    className="card__section card__successful-launch"
  >
    <strong>
      Successful Launch: 
    </strong>
    <span>
      <Component />
    </span>
  </div>
  <div
    className="card__section card__launch-landing"
  >
    <strong>
      Successful Landing: 
    </strong>
    <span>
      <Component />
    </span>
  </div>
</div>
`;

exports[`Filter should be possible to activate button with Spacebar 1`] = `
<Filter
  clickHandler={[Function]}
  isSelected={true}
  text="2007"
  value="2007"
>
  <button
    className="filter-btn selected"
    onClick={[Function]}
  >
    2007
  </button>
</Filter>
`;

exports[`Filter should render correctly with given props 1`] = `
<button
  className="filter-btn selected"
  onClick={[Function]}
>
  2007
</button>
`;

exports[`Footer should render correctly with no props 1`] = `
<footer
  className="footer"
>
  <span
    className="footer__text"
  >
    <strong>
      Developed By : 
    </strong>
    Suvechhya Banerjee
  </span>
</footer>
`;

exports[`Header should render correctly with no props 1`] = `
<header
  className="header"
>
  <h1>
    SpaceX Launch Programs
  </h1>
</header>
`;

exports[`Loader should render correctly with no props 1`] = `
<div
  className="loader"
>
  <div
    className="loader__content"
  />
</div>
`;
